server/services/deviceWearerService.test.ts

```diff
@@ -1,115 +1,150 @@
-  const mockRestClient = new RestClient(
-    'Mock Device Wearer Client',
-    config.apis.deviceWearer,
-    null,
-  ) as jest.Mocked<RestClient>
+  let deviceWearerService: DeviceWearerService
+  let fakeDataPlatformApi: nock.Scope
+  const accessToken = ''
+  const dummyData: DeviceWearer[] = [
+    {
+      deviceWearerId: '123456789',
+      firstName: 'J',
+      lastName: 'Smith',
+      type: 'Curfew',
+    },
+    {
+      deviceWearerId: '987654321',
+      firstName: 'J',
+      lastName: 'Doe',
+      type: 'Inclusion Zone',
+    },
+  ]
+
+  beforeEach(() => {
+    fakeDataPlatformApi = nock(config.apis.deviceWearer.url)
+    deviceWearerService = new DeviceWearerService()
+  })
 
   describe('findOne', () => {
     it('retrieves one device wearer when deviceWearerId exists', async () => {
       const testId = '987654321'
-      jest.spyOn(mockRestClient, 'get').mockImplementation(async input => {
-        if (input.path === `/device-wearers/v1/id/${testId}`) {
-          return { error: '', deviceWearers: [dummyData[1]] }
-        }
-        throw new Error('Wrong call to API')
+      const endpointUrl = `/device-wearers/v1/id/${testId}`
+      const expected: DeviceWearer = dummyData[1]
+      let called = 0
+
+      fakeDataPlatformApi.get(endpointUrl).reply(200, () => {
+        called += 1
+        return { error: '', deviceWearers: [dummyData[1]] }
       })
-      deviceWearerService = new DeviceWearerService(mockRestClient)
 
       const result = await deviceWearerService.findOne(accessToken, testId)
 
-      expect(result.deviceWearers[0].deviceWearerId).toEqual(testId)
-      expect(result.deviceWearers[0]).toEqual(dummyData[1])
+      expect(result.deviceWearerId).toEqual(testId)
+      expect(result).toEqual(expected)
+      expect(fakeDataPlatformApi.isDone()).toBeTruthy()
+      expect(called).toBe(1)
     })
 
     it('returns "No user found" error when deviceWearerId does not exist', async () => {
-      const testId = 'this is a totally fake ID'
-      const expectedError = 'No user found'
-      jest.spyOn(mockRestClient, 'get').mockImplementation(async input => {
-        if (input.path === `/device-wearers/v1/id/${testId}`) {
-          return { error: expectedError, deviceWearers: [] }
-        }
-        throw new Error('Wrong call to API')
+      const testId = 'This-is-a-totally-fake-Id'
+      const endpointUrl = `/device-wearers/v1/id/${testId}`
+      const expectedError = `No user found with ID ${testId}`
+      const expected = new Error(expectedError)
+      let called = 0
+
+      fakeDataPlatformApi.get(endpointUrl).reply(200, () => {
+        called += 1
+        return { error: '', deviceWearers: [], message: 'No data found' }
+      })
+
+      const result = deviceWearerService.findOne(accessToken, testId)
+
+      await expect(result).rejects.toMatchObject(expected)
+      expect(fakeDataPlatformApi.isDone()).toBeTruthy()
+      expect(called).toBe(1)
```

Comment:
When you create an expectation with nock `.isDone()` will only resolve to `true` if all the expectations have been requested. If the code being tested calls the API more times than there were created expectations, I think you should get a `Not Found` error?

Reply:
Not completely following - are you saying that line 69 `expect(called).toBe(1)` is unnecessary?

Reply:
I think so yeah